### DISCUSSION TITLE
when I use the extract_text funtion, the x_tolerance argument doesn't work for me.

### DISCUSSION THREAD

**Discussion by papandadj** [jsvine/pdfplumber#1010] [Ask for help with specific PDFs]
Thank you for providing such a great open-source project. However, there are some things I don't quite understand when using it.
## Describe the bug
<img width="1031" alt="image" src="https://github.com/jsvine/pdfplumber/assets/25424898/cea7043f-6ce7-43a0-a9bb-3e06e00fdede">

in this pdf, two characters are enclosed by red rectangles. 

Below is the information of my debugging code.
<img width="1716" alt="image" src="https://github.com/jsvine/pdfplumber/assets/25424898/c7bb5150-54df-4ea4-85f8-45fe25772486">

I have a question. The difference between the x1 of 'e' and the x0 of '的' is less than x_tolerance (364-361<10000). Why is there a space printed between the two characters?

Did I misunderstand something somewhere?

**Comment by jsvine** [comment]
Hi @papandadj, and thanks for your interest in this library. Without access to the specific PDF, it's difficult to provide an answer with certainty. But if I'm understanding your question correctly, this might help: In `pdfplumber`, literal whitespace characters (which seems to be what you're pointing to as `page.chars[49]`) are handled literally and are not affected by the tolerance parameters. If you don't want the whitespace characters to be used, I'd suggest filtering the page before extracting the text:

```python
without_whitespace = page.filter(lambda x: x.get("text") != "")
print(without_whitespace.extract_text())
```

**Reply by papandadj** [reply]
Think you for your answer. You just pointed out that is a literal space. Is the literal space directly scanned or calculted throht x_tolerace?

**Reply by jsvine** [reply]
By "literal space," I mean an explicit PDF instruction to print a space character (just like it contains explicit instructions to print the letters "K" "u" "b" etc.). This contrasts with implicit whitespace, where the PDF prints two letters some distance apart from one another, creating the *appearance* of a space. (`x_tolerance` is meant to provide more control over the handling of those implicit spaces.)

**Reply by papandadj** [reply]
Is there a concept of "literal line break", similar to the explicit instruction of '\n'?

**Reply by jsvine** [reply]
Not that I'm aware of. (And I don't think I've seen any in the wild.)

**Reply by papandadj** [reply]
I know what to do now. Think you.
